tensorboard/plugins/pr_curve/tf_pr_curve_dashboard/tf-pr-curve-card.html

```diff
@@ -0,0 +1,301 @@
+        // The value field is a function that returns a function because Polymer
+        // will actually call the value field if the field is a function. However,
+        // we actually want the value itself to be a function.
+        _xComponentsCreationMethod: {
+          type: Object,
+          readOnly: true,
+          value: () => (() => {
+            const scale = new Plottable.Scales.Linear();
+            return {
+              scale: scale,
+              axis: new Plottable.Axes.Numeric(scale, 'bottom'),
+              accessor: d => d.recall,
+            };
+          }),
+        },
+        _yValueAccessor: {
+          type: Object,
+          readOnly: true,
+          // This function returns a function because polymer calls the outer
+          // function to compute the value. We actually want the value of this
+          // property to be the inner function.
+          value: () => (d => d.precision),
+        },
+        _tooltipColumns: {
+          type: Array,
+          readOnly: true,
+          value: [
+            {
+              title: 'Name',
+              computeFunction: (d) => d.dataset.metadata().name,
+            },
+            {
+              title: 'Precision',
+              computeFunction: (d) => d.datum.precision,
+            },
+            {
+              title: 'Recall',
+              computeFunction: (d) => d.datum.recall,
+            },
+          ],
+        }
+      },
+      observers: [
+          'reload(run, tag)',
+          '_runsChanged(_attached, runs.*)',
+          '_setChartData(_runToPrCurveEntry)',
+      ],
+      _computeRunColor(run) {
+        return this._colorScaleFunction.scale(run);
+      },
+      attached() {
+        // Defer reloading until after we're attached, because that ensures that
+        // the requestManager has been set from above. (Polymer is tricky
+        // sometimes)
+        this._attached = true;
+        this.reload();
+      },
+      reload() {
+        if (!this._attached) {
+          return;
+        }
+        this._canceller.cancelAll();
+        const router = getRouter();
+        let url = router.pluginRoute('pr_curves', '/pr_curves');
+        url += url.indexOf('?') > -1 ? '&' : '?';
+        const urlRunsPortion = this.runs.map(r => `&run=${r}`).join('');
+        url += `tag=${this.tag}${urlRunsPortion}`;       
+        const updateData = this._canceller.cancellable(result => {
+          if (result.cancelled) {
+            return;
+          }
+
+          const runToData = result.value;
+          this.set('_runToData', runToData);
+          this.fire('data-updated', {
+            'runToData': runToData,
+          });
+        });
+        this.requestManager.request(url).then(updateData);
+      },
+      _setChartData(actualPrCurveEntryPerRun) {
+        _.forOwn(actualPrCurveEntryPerRun, (entry, run) => {
+          // Reverse the values so they are plotted in order, which allows for
+          // tool tips.
```

Comment:
Not clear to me what this comments means. How does reversing the values "allow for tooltips"? (Do you mean "causes the tooltips to render in the correct order", maybe?)